### PR TITLE
Converts GroupDAO to jooq

### PR DESCRIPTION
**Pull request by alokmenghrajani**

### CODE REVIEW

File: server/src/main/java/keywhiz/service/daos/GroupDAO.java [hunk 1]
```diff
@@ -16,40 +16,62 @@
...
+    r.setCreatedby(creator);
+    r.setUpdatedby(creator);
+    r.setDescription(description.orElse(null));
+    r.store();
+
+    return r.getId();
+  }
+
+  public void deleteGroup(Group group) {
+    GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.ID.eq(Math.toIntExact(group.getId())));
```
Comment (sul3n3t) [inline server/src/main/java/keywhiz/service/daos/GroupDAO.java]: Do you want to switch this to the same jooq pattern as ClientDAO?

Comment (alokmenghrajani) [inline server/src/main/java/keywhiz/service/daos/GroupDAO.java]: Yes.

File: server/src/main/java/keywhiz/service/daos/GroupDAO.java [hunk 2]
```diff
@@ -16,40 +16,62 @@
...
+    r.delete();
+  }
+
+  public Optional<Group> getGroup(String name) {
+    GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.NAME.eq(name));
+    if (r != null) {
+      return Optional.of(r.map(new GroupMapper()));
```
Comment (sul3n3t) [inline server/src/main/java/keywhiz/service/daos/GroupDAO.java]: `Optional.map`

File: server/src/main/java/keywhiz/service/daos/GroupDAO.java [hunk 3]
```diff
@@ -16,40 +16,62 @@
...
+    }
+    return Optional.empty();
+  }
+
+  public Optional<Group> getGroupById(long id) {
+    GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.ID.eq(Math.toIntExact(id)));
+    if (r != null) {
+      return Optional.of(r.map(new GroupMapper()));
```
Comment (sul3n3t) [inline server/src/main/java/keywhiz/service/daos/GroupDAO.java]: `Optional.map`

**Comment by sul3n3t** [comment]
:+1:

**Pull request merged**